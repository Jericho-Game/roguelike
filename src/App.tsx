import {
  Route,
  Routes,
} from 'react-router-dom';
import { useDispatch } from 'react-redux';

import { storeUser } from './store/user';
import ErrorBoundaryWrapper from './components/ErrorBoundaryWrapper';
import Layout from './components/Layout';
import routes from './routes';
import withOAuthCheck from './hocs/with-oauth-check';

function App() {
  const dispatch = useDispatch();
  if (globalThis?.localStorage?.getItem('userAuthorized')) {
    dispatch(storeUser());
  }
  return (
    <Layout>
      <ErrorBoundaryWrapper>
        <Routes>
          {routes.map(({ fetchData, ...routeProps }) => (
            <Route key={routeProps.path} {...routeProps} />
          ))}
        </Routes>
      </ErrorBoundaryWrapper>
    </Layout>
  );
}

export default withOAuthCheck(App);
