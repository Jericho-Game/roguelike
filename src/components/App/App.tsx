import {
  Route,
  Routes,
} from 'react-router-dom';
import { useDispatch } from 'react-redux';

import IndexPage from '../../pages/index';
import ForumPage from '../../pages/forum';
import LeaderboardPage from '../../pages/leaderboard';
import SignInPage from '../../pages/signin';
import SignUpPage from '../../pages/signup';
import ProfilePage from '../../pages/profile';
import Page404 from '../../pages/404';

import { storeUser } from '../../store/user';
import ErrorBoundaryWrapper from '../ErrorBoundaryWrapper';
import Layout from '../Layout';

export default function App() {
  const dispatch = useDispatch();

  if (window.localStorage.getItem('userAuthorized')) {
    dispatch(storeUser());
  }

  return (
    <Layout>
      <ErrorBoundaryWrapper>
        <Routes>
          <Route path="/" element={<IndexPage />} />
          <Route path="/forum" element={<ForumPage />} />
          <Route path="/forum/:id" element={<ForumPage />} />
          <Route path="/leaderboard" element={<LeaderboardPage />} />
          <Route path="/signin" element={<SignInPage />} />
          <Route path="/signup" element={<SignUpPage />} />
          <Route path="/profile" element={<ProfilePage />} />
          <Route path="*" element={<Page404 />} />
        </Routes>
      </ErrorBoundaryWrapper>
    </Layout>
  );
}
