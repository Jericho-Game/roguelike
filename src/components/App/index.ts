import { compose } from 'redux';
import App from './App';
import withAuthCheck from '../../hocs/with-auth-check';
import withOAuthCheck from '../../hocs/with-oauth-check';

export default compose(
  withOAuthCheck,
  withAuthCheck,
)(App);
