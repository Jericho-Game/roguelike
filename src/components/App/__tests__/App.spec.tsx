import { render } from '@testing-library/react';
import App from '../App';

type MockComponentProps = { children?: JSX.Element, element?: JSX.Element };

const dispatch = jest.fn();

jest.mock('../../../pages/index', () => ({
  __esModule: true,
  default: () => 'IndexPage',
}));
jest.mock('../../../pages/forum', () => ({
  __esModule: true,
  default: () => 'ForumPage',
}));
jest.mock('../../../pages/leaderboard', () => ({
  __esModule: true,
  default: () => 'LeaderboardPage',
}));
jest.mock('../../../pages/signin', () => ({
  __esModule: true,
  default: () => 'SignInPage',
}));
jest.mock('../../../pages/signup', () => ({
  __esModule: true,
  default: () => 'SignUpPage',
}));
jest.mock('../../../pages/profile', () => ({
  __esModule: true,
  default: () => 'ProfilePage',
}));
jest.mock('../../../pages/404', () => ({
  __esModule: true,
  default: () => 'Page404',
}));

jest.mock('../../ErrorBoundaryWrapper', () => ({
  __esModule: true,
  default: ({ children }: MockComponentProps) => <div data-testid="ErrorBoundaryWrapper">{children}</div>,
}));
jest.mock('../../Layout', () => ({
  __esModule: true,
  default: ({ children }: MockComponentProps) => <div data-testid="Layout">{children}</div>,
}));

jest.mock('react-redux', () => ({
  __esModule: true,
  useDispatch: () => dispatch,
}));

jest.mock('react-router-dom', () => ({
  __esModule: true,
  BrowserRouter: ({ children, ...props }: MockComponentProps) => <div data-testid="BrowserRouter" {...props}>{children}</div>,
  Routes: ({ children, ...props }: MockComponentProps) => <div data-testid="Routes" {...props}>{children}</div>,
  Route: ({ element, ...props }: MockComponentProps) => <div data-testid="Route" {...props}>{element}</div>,
}));

describe('App Test', () => {
  it('matches snapshot', () => {
    const { container } = render(<App />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('has IndexPage', () => {
    const { container } = render(<App />);
    const indexPageRoute = container.querySelector("[path='/']");

    expect(indexPageRoute).toBeInTheDocument();
    expect(indexPageRoute?.textContent).toBe('IndexPage');
  });

  it('has ForumPage', () => {
    const { container } = render(<App />);
    const forumPageRoute = container.querySelector("[path='/forum']");
    const forumIdPageRoute = container.querySelector("[path='/forum/:id']");

    expect(forumPageRoute).toBeInTheDocument();
    expect(forumIdPageRoute).toBeInTheDocument();
    expect(forumPageRoute?.textContent).toBe('ForumPage');
    expect(forumIdPageRoute?.textContent).toBe('ForumPage');
  });

  it('has LeaderboardPage', () => {
    const { container } = render(<App />);
    const leaderboardPageRoute = container.querySelector("[path='/leaderboard']");

    expect(leaderboardPageRoute).toBeInTheDocument();
    expect(leaderboardPageRoute?.textContent).toBe('LeaderboardPage');
  });

  it('has SignInPage', () => {
    const { container } = render(<App />);
    const signInPageRoute = container.querySelector("[path='/signin']");

    expect(signInPageRoute).toBeInTheDocument();
    expect(signInPageRoute?.textContent).toBe('SignInPage');
  });

  it('has SignUpPage', () => {
    const { container } = render(<App />);
    const signUpPageRoute = container.querySelector("[path='/signup']");

    expect(signUpPageRoute).toBeInTheDocument();
    expect(signUpPageRoute?.textContent).toBe('SignUpPage');
  });

  it('has ProfilePage', () => {
    const { container } = render(<App />);
    const profilePageRoute = container.querySelector("[path='/profile']");

    expect(profilePageRoute).toBeInTheDocument();
    expect(profilePageRoute?.textContent).toBe('ProfilePage');
  });

  it('has Page404', () => {
    const { container } = render(<App />);
    const notFoundPageRoute = container.querySelector("[path='*']");

    expect(notFoundPageRoute).toBeInTheDocument();
    expect(notFoundPageRoute?.textContent).toBe('Page404');
  });
});
