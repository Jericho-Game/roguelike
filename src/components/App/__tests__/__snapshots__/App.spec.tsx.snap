// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Test matches snapshot 1`] = `
<div
  data-testid="BrowserRouter"
>
  <div
    data-testid="Layout"
  >
    <div
      data-testid="ErrorBoundaryWrapper"
    >
      <div
        data-testid="Routes"
      >
        <div
          data-testid="Route"
          path="/"
        >
          IndexPage
        </div>
        <div
          data-testid="Route"
          path="/forum"
        >
          ForumPage
        </div>
        <div
          data-testid="Route"
          path="/forum/:id"
        >
          ForumPage
        </div>
        <div
          data-testid="Route"
          path="/leaderboard"
        >
          LeaderboardPage
        </div>
        <div
          data-testid="Route"
          path="/signin"
        >
          SignInPage
        </div>
        <div
          data-testid="Route"
          path="/signup"
        >
          SignUpPage
        </div>
        <div
          data-testid="Route"
          path="/profile"
        >
          ProfilePage
        </div>
        <div
          data-testid="Route"
          path="*"
        >
          Page404
        </div>
      </div>
    </div>
  </div>
</div>
`;
